lib/galaxy/managers/notification.py

```diff
@@ -0,0 +1,100 @@
+from typing import (
+    Any,
+    Dict,
+    List,
+)
+
+from sqlalchemy import false
+
+from galaxy import model
+from galaxy.exceptions import (
+    Conflict,
+    ObjectAttributeMissingException,
+    ObjectNotFound,
+)
+from galaxy.managers.base import decode_id
+from galaxy.managers.context import ProvidesAppContext
+from galaxy.schema.fields import EncodedDatabaseIdField
+from galaxy.structured_app import MinimalManagerApp
+from galaxy.web import url_for
+
+
+class NotificationManager:
```

Comment:
This is looking good, but can you remove all id encoding and decoding from the Manager ? I don't think we need this at all for a modern API. This should all just work with integers.